code/modules/power/singularity/singularity.dm

```diff
@@ -78,7 +78,10 @@
 	consume(user)
 
 /obj/singularity/attackby(obj/item/W, mob/user, params)
-	consume(user)
+	if(istype(W, /obj/item/gun/ballistic/SRN_rocketlauncher))
```

Comment:
I don't believe this will work at all.
This will either only work if directly attacked by the rocket launcher (smack it with the launcher), or work even on a missed shot.

You may need to attach the user to the projectile itself, then give the user the achievement upon the creation of the rift.